packages/office-ui-fabric-react/src/components/SpinButton/SpinButton.tsx

```diff
@@ -0,0 +1,398 @@
+            < Label
+              id={ this._labelId }
+              htmlFor={ this._inputId }
+              className='ms-SpinButtonLabel'>{ label }
+            </Label> }
+        </div> }
+        <div className={ css('ms-SpinButtonWrapper', ((labelPosition === Position.top || labelPosition === Position.bottom) ? 'topBottom' : '')) }>
+          <input
+            value={ value }
+            id={ this._inputId }
+            onChange={ this._onChange }
+            onInput={ this._onInputChange }
+            className={ css('ms-SpinButton-Input', (disabled ? ' disabled' : '')) }
+            type='text'
+            role='spinbutton'
+            aria-labelledby={ label && this._labelId }
+            aria-valuenow={ value }
+            aria-valuemin={ min && String(min) }
+            aria-valuemax={ max && String(max) }
+            onBlur={ this._validate }
+            ref={ this._resolveRef('_input') }
+            onFocus={ this.focus }
+            onKeyDown={ this._handleKeyDown }
+            onKeyUp={ this._handleKeyUp }
+            readOnly={ disabled }
+            aria-disabled={ disabled }
+          />
+          <span className='ms-ArrowBox'>
+            <IconButton
+              className={ css('ms-UpButton', (keyboardSpinDirection === KeyboardSpinDirection.up ? 'active' : '')) }
+              disabled={ disabled }
+              iconProps={ incrementButtonIcon }
+              title='Increase'
+              aria-hidden='true'
+              onMouseDown={ () => { this._updateValue(true /* shouldSpin */, this._onIncrement); } }
+              onMouseLeave={ this._stop }
+              onMouseUp={ this._stop }
+              tabIndex={ -1 }
+            />
+            <IconButton
+              className={ css('ms-DownButton', (keyboardSpinDirection === KeyboardSpinDirection.down ? 'active' : '')) }
+              disabled={ disabled }
+              iconProps={ decrementButtonIcon }
+              title='Decrease'
+              aria-hidden='true'
+              onMouseDown={ () => { this._updateValue(true /* shouldSpin */, this._onDecrement); } }
+              onMouseLeave={ this._stop }
+              onMouseUp={ this._stop }
+              tabIndex={ -1 }
+            />
+          </span >
+        </div >
+        { labelPosition === Position.bottom && <div className='ms-labelWrapper' style={ this._labelDirectionHelper() }>
+          { iconProps && <i className={ css('ms-SpinButtonIcon', 'ms-Icon', ('ms-Icon--' + iconProps.iconName)) } aria-hidden='true'></i> }
+          { label &&
+            <Label
+              id={ this._labelId }
+              htmlFor={ this._inputId }
+              className='ms-SpinButtonLabel'>{ label }
+            </Label>
+          }
+        </div> }
+      </ div >
+    );
+  }
+
+  /**
+   * OnFocus select the contents of the input
+   */
+  public focus() {
+    if (this._spinning || this.state.keyboardSpinDirection !== KeyboardSpinDirection.notSpinning) {
```

Comment:
We updated `_spinning` to `_spinningByMouse`. They're both needed, because one refers to spinning via the keyboard, and the other one via the mouse and they can potentially collide (which we should handle gracefully).